### PR TITLE
[KYUUBI #6615] Make Jetty sending server version in response configurable

### LINKED ISSUE #6615: [FEATURE] Kyuubi Jetty server should be able to hide version info in response, thus avoiding vulnerability exposure.
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

Add a switch for enabling/disabling jetty sending server version in response.

### Motivation

By default Jetty always send version in ressponse, such as Jetty(9.4.54.v20240208), which could expose vulnerablilty. Malicious visitors could easily detect Jetty version and then perform specific attack.

### Describe the solution

Add a config item that controls whether Jetty should send its version in response.

Sending Jetty version could be disabled by calling HttpConfiguration::setSendServerVersion(false)

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by paul8263**
# :mag: Description
## Issue References 🔗

This pull request fixes #6615

## Describe Your Solution 🔧

Add a config item that controls whether Jetty should send its version in response.

This is an additional patch which enables/disables sending Jetty version for  prometheus reporter.

Sending Jetty version could be disabled by calling HttpConfiguration::setSendServerVersion(false)

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Compiled and tested manually.

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala [hunk 1]
```diff
@@ -35,12 +36,21 @@ class PrometheusReporterService(registry: MetricRegistry)
...
   // VisibleForTesting
   private[metrics] var httpServer: Server = _
+  private[metrics] var connector: ServerConnector = _
   @volatile protected var isStarted = false
 
   override def initialize(conf: KyuubiConf): Unit = {
     val port = conf.get(MetricsConf.METRICS_PROMETHEUS_PORT)
     val contextPath = conf.get(MetricsConf.METRICS_PROMETHEUS_PATH)
-    httpServer = new Server(port)
+    val sendServerVersion = conf.get(FRONTEND_JETTY_SEND_VERSION_ENABLED)
```
Comment (bowenliang123) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala:45]:
```suggestion
    val jettyVersionEnabled = conf.get(FRONTEND_JETTY_SEND_VERSION_ENABLED)
```

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala [hunk 2]
```diff
@@ -35,12 +36,21 @@ class PrometheusReporterService(registry: MetricRegistry)
 
   // VisibleForTesting
   private[metrics] var httpServer: Server = _
+  private[metrics] var connector: ServerConnector = _
```
Comment (bowenliang123) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala:39]:
```suggestion
  private[metrics] var httpServerConnector: ServerConnector = _
```
To follow the naming conversion with existed `httpServer` variable.

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala [hunk 3]
```diff
@@ -79,11 +90,13 @@ class PrometheusReporterService(registry: MetricRegistry)
     if (httpServer != null) {
       try {
         httpServer.stop()
+        connector.stop()
```
[the remark this answers is not in the document]
Comment (paul8263) [reply kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala]: Hi @bowenliang123 ,
Thanks for the review.
I refered to the codes in JettyServer, in which the stop method closes the server first and then the server conenctor. I think it should work as well.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Sorry missing this PR, merged to master

**Comment by paul8263** [comment]
> Sorry missing this PR, merged to master

Thanks bro.

**Pull request closed without merging**